cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -663,6 +675,104 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
     return cw.toByteArray();
   }
 
+  /**
+   * Rewrites the temporary file name used for checkpoints from checkpointDir/temp to
+   * checkpointDir/temp-(checkpoint_time_milliseconds).
+   *
+   * @param byteCodeStream {@link InputStream} for reading in the original bytecode.
+   * @return the rewritten bytecode
+   */
+  private byte[] rewriteTempFileNameForCheckpoint(InputStream byteCodeStream) throws IOException {
+    String checkpointTimeFieldName =
+      "org$apache$spark$streaming$CheckpointWriter$CheckpointWriteHandler$$checkpointTime";
+    Type stringType = Type.getType(String.class);
+    Type stringBuilderType = Type.getType(StringBuilder.class);
+
+    ClassReader cr = new ClassReader(byteCodeStream);
+    ClassWriter cw = new ClassWriter(0);
+
+    cr.accept(new ClassVisitor(Opcodes.ASM5, cw) {
+      @Override
+      public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+
+        MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
+
+        return new GeneratorAdapter(Opcodes.ASM5, mv, access, name, desc) {
+
+          boolean tempStringAddedToStack = false;
+          boolean hasCheckpointTimeField = false;
+          final Method hadoopPathConstructorMethod =
+            new Method("<init>", Type.VOID_TYPE, new Type[]{ stringType, stringType });
+
+          @Override
+          public void visitFieldInsn(int opcode, String owner, String name, String descriptor) {
```

Comment:
You shouldn't be override this. This get called when a field is used inside a method. What I suggested earlier is to override the ClassVisitor.visitField method to make the field is defined in the class.

Reply:
Oh, I see what you mean now. I'll move this logic into the ClassVisitor instead